File: TreeViewAdvDemo/VB/CustomTreeNodeAdv.vb
Imports Syncfusion.Windows.Forms.Tools

Namespace TreeViewAdvNodeId
    'Creating custom class
    Public Class CustomTreeNodeAdv
        Inherits TreeNodeAdv
        Public Sub New()
            'do something
        End Sub

        Private m_NodeId As Integer
        ''' <summary>
        ''' Gets/Sets the Id in TreeNodeAdv
        ''' </summary>
        Public Property ID() As Integer
            Get
                Return m_NodeId
            End Get
            Set(ByVal value As Integer)
                m_NodeId = value
            End Set
        End Property
    End Class
End Namespace

File: TreeViewAdvDemo/C#/CustomTreeNodeAdv.cs
using Syncfusion.Windows.Forms.Tools;

namespace TreeViewAdvNodeId
{
    //Creating custom class
    public class CustomTreeNodeAdv : TreeNodeAdv
	{
		public CustomTreeNodeAdv()
		{
			//do something
		}

		private int m_NodeId;
		/// <summary>
		/// Gets/Sets the Id in TreeNodeAdv
		/// </summary>
		public int ID
		{
			get
			{
				return m_NodeId;
			}
			set
			{
				m_NodeId = value;
			}
		}
	}
}